client/pcr.go

```diff
@@ -81,6 +121,54 @@ func ReadPCRs(rw io.ReadWriter, sel tpm2.PCRSelection) (*pb.PCRs, error) {
 	return &pl, nil
 }
 
+// readPcrHelper fetchs the digest of a single PCR index.
+func readPcrHelper(thetpm transport.TPM, hash tpm.AlgID, index int) (*tpm2b.Digest, error) {
+	pcrRead := tpm2Direct.PCRRead{
+		PCRSelectionIn: tpml.PCRSelection{
+			PCRSelections: []tpms.PCRSelection{
+				{
+					Hash:      hash,
+					PCRSelect: make([]byte, 3),
+				},
+			},
+		},
+	}
+
+	pcrRead.PCRSelectionIn.PCRSelections[0].PCRSelect[index/8] = 1 << (index % 8)
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to pcrRead")
+	}
+
+	return &pcrReadRsp.PCRValues.Digests[0], nil
+}
+
+// ReadPCRsDirect fetches all the PCR values specified in sel, making multiple calls
+// to the TPM if necessary.
+func ReadPCRsDirect(thetpm transport.TPM, sel tpms.PCRSelection) (*pb.PCRs, error) {
```

Comment:
should the interface to `ReadPCRsDirect` be `tpms.PCRSelection`? I wonder if a slice of ints and a hash algo would be more user friendly. Otherwise, users of the library would have to do bit manipulation.

We can also consider making this private for now if you don't want to tackle that in this PR.

Reply:
Another option is to have a helper that converts an int slice into the tpmdirect `PCRSelect` slice

Reply:
Currently, this is why we must have something like CreatePCRSelection public. PCRSelections are used outside of reading so perhaps having a function that will just made the tpms.PCRSelection available to the user would be wise.

Reply:
I think we should make this, `ReadAllPCRsDirect`, `FullPcrSelDirect`, and `CreatePCRSelection` private for now, and handle how we want this to be exported in a followup PR.

Reply:
+1, we don't have to solve this now so private seems best

Reply:
I have made it all private but `FullPcrSelDirect` as it is used in the client_testing package.